app/src/main/java/de/westnordost/streetcomplete/quests/building_type/AddBuildingType.kt

```diff
@@ -37,7 +37,10 @@ class AddBuildingType (o: OverpassMapDataAndGeometryApi) : SimpleOverpassQuestTy
             changes.delete("building")
             changes.add("man_made", manMade)
         } else {
-            changes.modify("building", answer)
+            if (answer.contains("historic") || answer.contains("abandoned") || answer.contains("ruin"))
+                changes.add(answer, "yes")
+            else
+                changes.modify("building", answer)
         }
```

Comment:
You could pull up that else. I.e. if ... else if... else

Also, why `answer.contains("historic")` and not `answer == "historic"`?